### PR TITLE
Don't use iptc.easy for deleting rules.

### LINKED ISSUE #76: Don't use iptc.easy for port/network blocking
Deleting a rule fails in 50% of my test runs because it can't find a rule to delete.

Use a "non-easy" API for that:
```
import iptc

tbl = iptc.Table('filter')
ch = iptc.Chain(tbl, 'INPUT')
for r in ch.rules:
    for m in r.matches:
        print(m.get_all_parameters())
        if m.comment == 'added by WoTT':
            print('deleting')
            ch.delete_rule(r)
            break
```

### PR DESCRIPTION
**Pull request by a-martynovich**
Fixes #76

### CODE REVIEW

File: agent/security_helper.py [hunk 1]
```diff
@@ -38,7 +41,12 @@ def is_firewall_enabled():
 
 def get_firewall_rules():
     """Get all FILTER table rules"""
-    return iptc_helper.dump_table('filter')
+    res = {}
```
Comment (rptrchv) [inline agent/security_helper.py]: Pls use more meaningful names

File: agent/security_helper.py [hunk 2]
```diff
@@ -53,7 +61,7 @@ def netstat_scan():
         'remote_address': c.raddr,
         'status': c.status if c.type == socket.SOCK_STREAM else None,
         'pid': c.pid
-    } for c in connections]
+    } for c in connections if c.raddr and c.laddr]
```
Comment (rptrchv) [inline agent/security_helper.py]: Probably we shouldn't skip connections with empty local address info. They still can be useful for blocking remote connections

File: agent/security_helper.py [hunk 3]
```diff
@@ -131,21 +139,30 @@ def prepare_iptables():
...
-            iptc_helper.delete_rule(table, chain, r)
+    iptc_helper.batch_begin()
+    tbl = iptc.Table(table)
+    tbl.autocommit = False
+    ch = iptc.Chain(tbl, chain)
+
+    for r in ch.rules:
+        for m in r.matches:
+            if m.comment == WOTT_COMMENT:
+                print('deleting')
```
Comment (rptrchv) [inline agent/security_helper.py]: is it ok leaving `print` statements?

Comment (a-martynovich) [reply agent/security_helper.py]: no, it's not

File: agent/security_helper.py (hunk 2 above)
Comment (a-martynovich) [reply agent/security_helper.py]: @rptrchv if API can handle it now, I'll revert this

Comment (rptrchv) [reply agent/security_helper.py]: yes, it can

File: agent/security_helper.py [hunk 4]
```diff
@@ -61,7 +61,7 @@ def netstat_scan():
         'remote_address': c.raddr,
         'status': c.status if c.type == socket.SOCK_STREAM else None,
         'pid': c.pid
-    } for c in connections if c.raddr and c.laddr]
+    } for c in connections]
```
Comment (rptrchv) [inline agent/security_helper.py]: you've stripped out too much - the check for `c.raddr` not empty is still required

Comment (a-martynovich) [reply agent/security_helper.py]: @rptrchv But API server now handles this case, doesn't it?
In the future we won't use nmap and instead we'll get open ports from netstat report. This is where raddr will be empty.

Comment (rptrchv) [reply agent/security_helper.py]: > But API server now handles this case, doesn't it?
yes, but there's no need to transmit useless data

Comment (a-martynovich) [reply agent/security_helper.py]: @rptrchv it's not useless, as I've said above.

**Pull request merged**